nilearn/maskers/nifti_spheres_masker.py

```diff
@@ -318,8 +322,183 @@ def __init__(
         self.memory = memory
         self.memory_level = memory_level
 
+        # Parameters for reporting
+        self.reports = reports
+        self.report_id = -1
```

Comment:
??

Reply:
Now report id is set in the report content and will be incremented every time generate_report is called. This was adapted directly from nifti_maps_masker